[SeanChinJunKai] iP

SeanChinJunKai opened the pull request:
# Duke
Duke frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- -~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/SeanChinJunKai/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉 

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)


If you Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class,args);
    }
}
```

Commit:
Add Gradle support

SeanChinJunKai committed:
Updated gitignore to exclude .class files

SeanChinJunKai committed:
Completed Level 1

SeanChinJunKai committed:
Edited console display by inserting tab

SeanChinJunKai committed:
Completed Level 2

SeanChinJunKai committed:
Completed Level 3

SeanChinJunKai committed:
Added getStatusIcon and removed id

SeanChinJunKai committed:
Completed Level 4

SeanChinJunKai committed:
Completed A-TextUiTesting

SeanChinJunKai committed:
Completed Level-5

SeanChinJunKai committed:
Completed Level-6

SeanChinJunKai committed:
Code reformatting

SeanChinJunKai committed:
Updated A-TextUITesting

SeanChinJunKai committed:
Complete Level-7

SeanChinJunKai committed:
Complete Level-8

SeanChinJunKai committed:
Merge branch 'branch-Level-8'

SeanChinJunKai committed:
Fix load deadline functionality

SeanChinJunKai committed:
Merge branch 'add-gradle-support'

SeanChinJunKai committed:
Add Checkstyle

SeanChinJunKai committed:
Add more OOP

SeanChinJunKai committed:
Remove unneeded files

SeanChinJunKai committed:
Organise classes into duke package

SeanChinJunKai committed:
Add JUnit Tests

SeanChinJunKai committed:
Add JavaDocs

SeanChinJunKai committed:
Tweak code to comply with coding standard

SeanChinJunKai committed:
Add search functionality

SeanChinJunKai committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Ui.java

SeanChinJunKai committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/TaskList.java
#	src/main/java/duke/Ui.java

SeanChinJunKai committed:
Update Javadocs

SeanChinJunKai committed:
Configure styling to match Gradle CheckStyle

SeanChinJunKai committed:
Refactor code to make full use of TaskList abstraction

SeanChinJunKai committed:
Refactor code for find functionality

SeanChinJunKai committed:
Add Level-10 functionality

SeanChinJunKai committed:
Update javadocs

SeanChinJunKai committed:
Edit JavaDocs Part 2

JamesLiuZX commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,177 @@
+            break;
+        }
+        default:
+            throw new DukeException("Invalid command entered!");
```
Good cases of exception handling! The exceptions can be more specific if time permits, but since this is a chatbot that focuses heavily on UI, there isn't really a need for too much exception handling either beyond printing error messages.

JamesLiuZX commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,105 @@
+ * @author Sean Chin Jun Kai
+ */
+public class Storage {
+    private static final String DIR_PATH = System.getProperty("user.home") + "/ip/data";
+    private String filePath;
```
Good job on setting the variables as private! Also for using final appropriately.

gekhro replied:
Good work on making the file paths private

JamesLiuZX commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,105 @@
+     * @return Task object.
+     * @throws DukeException if there is an error reading the file.
+     */
+    public static Task readTaskString(String input) throws DukeException {
```
This method could be under Parser, I think even though it is under the context of the text file it is still a parser method.

JamesLiuZX reviewed (commented):
Overall great coding standards! Can tell that you are an experienced programmer so I don't have much to add apart from exceptions and packaging which are not really coding standards. Keep up the good work!

gekhro commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,105 @@
+     * @return string to be displayed to user.
+     */
+
+    public String getResponse(String input) {
```
maybe shift the getResponse method to the command class, this could simplify code in duke.java. Enum list could be in the parser class as well

gekhro commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,65 @@
+ *
+ * @author Sean Chin Jun Kai
+ */
+public class Task {
```
good job on making javdocs for your methods

gekhro commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,103 @@
+     * @return chosen Task.
+     * @throws DukeException if given ID does not exist in TaskList.
+     */
+    public Task getTask(int taskNumber) throws DukeException {
```
good use of dukeException to convert the default array out of bounds exception to a custom one

SeanChinJunKai committed:
Update gradle build file

SeanChinJunKai committed:
Add basic assertions

SeanChinJunKai committed:
Refactor Duke class to adhere to code quality standards

Duke methods does not adhere to code quality standards.

Poor code quality can result in readers not being able to understand the code whilst making it more prone to bugs and errors.

Usage of blank lines in order to implement logical structuring of code in the getResponse method and the removal of redundant method calls in the same method.

By laying out the code such that it adheres to logical structure, this makes it very clear for readers to easily tell at a glance what we are doing. Furthermore, the removal of redundant code can reduce the complexity of code being read by readers allowing them to better understand our intentions as the developer.

SeanChinJunKai committed:
Revert "Refactor Duke class to adhere to code quality standards"

This reverts commit d0c80715f5149ab3c99e76e2790697f8db6c7565.

SeanChinJunKai committed:
Refactor Duke class to adhere to code quality standards.

Duke methods does not adhere to code quality standards.

Poor code quality can result in readers not being able to understand the code whilst making it more prone to bugs and errors.

Usage of blank lines in order to implement logical structuring of code in the getResponse method and the removal of redundant method calls in the same method.

By laying out the code such that it adheres to logical structure, this makes it very clear for readers to easily tell at a glance what we are doing. Furthermore, the removal of redundant code can reduce the complexity of code being read by readers allowing them to better understand our intentions as the developer.

SeanChinJunKai committed:
Refactor Parser class to adhere to code quality standards

Parser methods do not adhere to code quality standards.

Poor code quality can result in readers not being able to understand the code whilst making it more prone to bugs and errors.

Usage of blank lines in order to implement logical structuring of code whilst implementing guard clauses to make happy path prominent.

By laying out the code such that it adheres to logical structure, this makes it very clear for readers to easily tell at a glance what we are doing. Furthermore, the use of guard clauses prevent readers from getting distracted by alternative paths taken when error conditions happen

SeanChinJunKai committed:
Refactor Storage, TaskList and Test classes

SeanChinJunKai committed:
Merge pull request #1 from SeanChinJunKai/branch-A-Assertions

Add basic assertions

SeanChinJunKai committed:
Merge branch 'master' into branch-A-CodeQuality

SeanChinJunKai committed:
Merge pull request #2 from SeanChinJunKai/branch-A-CodeQuality

Adhering to code quality

SeanChinJunKai committed:
Fix missing curly braces

SeanChinJunKai committed:
Fix missing text before displaying tasks in the list

SeanChinJunKai committed:
Add duplicate task detection functionality

SeanChinJunKai committed:
Add abstraction for Duke commands

SeanChinJunKai committed:
Added product name to application window

SeanChinJunKai committed:
Update README.md to be the user guide
